File: game/state.py
# state.py

class GameState:
    def __init__(self, num_players=2):
        self.turn = 0
        self.current_player = 0
        self.num_players = num_players
        self.winner = None

    def __str__(self):
        return f"Turn: {self.turn}, Current Player: {self.current_player}"

File: game/state.ts
import { Player } from "./player"
import { World } from "../ecs/core";

export enum turnPhase{
  TURN_START = "TURN_START",
  MAIN_PHASE = "MAIN_PHASE",
  ATTACK_PHASE = "ATTACK_PHASE"
}

export class GameState {
  turn: number;
  currentPlayer: number;
  players: Player[];
  winner: number | null;
  turnPhase: turnPhase
  world: World;

  constructor() {
    this.turn = 0;
    this.currentPlayer = 0;
    this.players = [];
    this.winner = null;
    this.turnPhase = turnPhase.TURN_START;
    this.world = new World();
  }

  toString(): string {
    return `Turn: ${this.turn}, Current Player: ${this.currentPlayer}`;
  }
}